demo/ios/PorcupineDemo/ViewController.swift

```diff
@@ -0,0 +1,105 @@
+    @IBOutlet weak var startButton: UIButton!
+
+    let wakeWords = [
+        "Alexa", "Americano", "Avocado", "Blueberry", "Bumblebee", "Caterpillar", "Christina", "Dragonfly", "Flamingo",
+        "Francesca", "Grapefruit", "Grasshopper", "Iguana", "Picovoice", "Pineapple", "Porcupine", "Raspberry",
+        "Terminator", "Vancouver"]
+    var wakeWord = "Alexa"
+
+    var porcupineManager: PorcupineManager!
+    var isRecording = false
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        wakeWordPicker.delegate = self
+        wakeWordPicker.dataSource = self
+        
+        let viewSize = view.frame.size
+        let startButtonSize = CGSize(width: 120, height: 120)
+        startButton.frame.size = startButtonSize
+        startButton.frame.origin =
+            CGPoint(x: (viewSize.width - startButtonSize.width) / 2, y: viewSize.height - (startButtonSize.height + 40))
+        startButton.layer.cornerRadius = 0.5 * startButton.bounds.size.width
+        startButton.clipsToBounds = true
+
+        wakeWordPicker.frame.origin = CGPoint(x: 0, y: 40)
+        wakeWordPicker.frame.size = CGSize(width: viewSize.width, height: viewSize.height - startButtonSize.height - 120)
+    }
+
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return wakeWords.count
+    }
+
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return wakeWords[row]
+    }
+
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        wakeWord = wakeWords[row]
+    }
+
+    @IBAction func toggleStartButton(_ sender: UIButton) {
+        if !isRecording {
+            let modelFilePath = Bundle.main.path(forResource: "porcupine_params", ofType: "pv")
+            let keywordFilePath = Bundle.main.path(forResource: wakeWord.lowercased() + "_ios", ofType: "ppn")
+
+            let originalColor = self.view.backgroundColor
+            let keywordCallback: (() -> Void) = {
+                DispatchQueue.main.sync { self.view.backgroundColor = UIColor.orange }
+                DispatchQueue.main.asyncAfter(
+                    deadline: DispatchTime.now() + 1.0,
+                    execute: {self.view.backgroundColor = originalColor})
+            }
+
+            porcupineManager = PorcupineManager(
+                modelFilePath: modelFilePath!,
+                keywordFilePath: keywordFilePath!,
+                sensitivity: 0.5,
+                keywordCallback: keywordCallback)
+
+            do {
+                try porcupineManager.start()
+            } catch let ex {
+                let alert = UIAlertController(
+                    title: "Alert",
+                    message: "Failed to initialized Porcupine " + ex.localizedDescription,
```

Comment:
I'm not sure but just show something that is user-friendly if it is user-facing. Maybe something like "Something went wrong..."

Reply:
Agreed